### PR TITLE
cli: Align the handling of case-insensitive list options

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: cli/src/main/kotlin/commands/AnalyzerCommand.kt [hunk 1]
```diff
@@ -57,7 +57,8 @@ import org.ossreviewtoolkit.utils.perf
 import org.ossreviewtoolkit.utils.safeMkdirs
 
 class AnalyzerCommand : CliktCommand(name = "analyze", help = "Determine dependencies of a software project.") {
-    private val allPackageManagersByName = PackageManager.ALL.associateBy { it.managerName.toUpperCase() }
+    private val allPackageManagersByName = PackageManager.ALL.associateBy { it.managerName }
+        .toSortedMap(String.CASE_INSENSITIVE_ORDER)
```
Comment (KorSin) [inline cli/src/main/kotlin/commands/AnalyzerCommand.kt:61]: Just to be sure: Does this make the map actually `case-insensitive`, i.e., are "abc" and "ABC" hashed to the same value?

File: cli/src/main/kotlin/commands/ReporterCommand.kt [hunk 2]
```diff
@@ -251,7 +250,7 @@ class ReporterCommand : CliktCommand(
 
         reportOptions.forEach { (format, option) ->
```
Comment (KorSin) [inline cli/src/main/kotlin/commands/ReporterCommand.kt:251]: Couldn't you make this map also case-insensitive? But maybe this is rather for another PR.

File: cli/src/main/kotlin/commands/AnalyzerCommand.kt (hunk 1 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/AnalyzerCommand.kt:61]: Yes. I agree the name "CASE_INSENSITIVE_*ORDER*" is misleading, but in fact the keys become case-insensitive.

File: analyzer/src/main/kotlin/managers/Carthage.kt [hunk 3]
```diff
@@ -123,7 +123,7 @@ class Carthage(
     }
```
[the remark this answers is not in the document]
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Carthage.kt:123]: Yes, I was in need of doing something crazy today...

File: cli/src/main/kotlin/commands/ReporterCommand.kt (hunk 2 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/ReporterCommand.kt:252]: I was nicer to do than I thought, thanks to a new function in Kotlin 1.4, so done.

**Review by KorSin — APPROVED** [review]

**Review by KorSin — APPROVED** [review]

File: analyzer/src/main/kotlin/managers/Carthage.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (oheger-bosch) [reply analyzer/src/main/kotlin/managers/Carthage.kt:123]: Is there some evidence that this actually yields a performance gain? I usually prefer leaving such micro optimizations to the compiler or the runtime. But I am not against this change.

**Review by oheger-bosch — APPROVED** [review]

**Comment by sschuberth** [comment]
> Is there some evidence that this actually yields a performance gain?

Nope, hence the smiley. But I also did not really put any effort into researching this.

> I usually prefer leaving such micro optimizations to the compiler or the runtime. But I am not against this change.

Right, ideally the compiler should take care of such things. For me, this primarily is another alignment commit for the code base.

**Pull request merged**